Minesweeper.java

```diff
@@ -0,0 +1,99 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.util.*;
+import java.awt.event.*;
+
+public class Minesweeper {
+
+    // The size in pixels for the frame.
+    private static final int SIZE = 500;
+    private JFrame  frame;
+
+    private final ActionListener actionListener = actionEvent -> {
+        Object source = actionEvent.getSource();
+    };
+
+    private class Cell extends JButton {
+        private final int row;
+        private final int col;
+        private int value;
+
+        Cell(final int row, final int col,
+             final ActionListener actionListener) {
+            this.row = row;
+            this.col = col;
+            addActionListener(actionListener);
+            setText("");
+        }
+
+        //TODO: For later implementations
+        /*int getValue() {
+            return value;
+        }
+
+        void setValue(int value) {
+            this.value = value;
+        }
+
+        boolean isAMine() {
+            return value == MINE;
+        }*/
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (obj == null || getClass() != obj.getClass())
+                return false;
+            Cell cell = (Cell) obj;
+            return row == cell.row &&
+                   col == cell.col;
+        }
+
+        @Override
```

Comment:
this is never used, does it need to be here?

Reply:
thats just standard for the objects, i overried equals and hash so we can compare two objects by their values and not the pointers, it was the same with the madtrucker objects